### PR TITLE
[DC-1319] RDR eval: Add check for basics survey module

### PR DESCRIPTION
**Pull request by dcampbell-vumc**
* Added code to check that participants with data in other survey modules also have data from the basics survey module to the RDR notebook

### CODE REVIEW

File: data_steward/analytics/cdr_ops/rdr_export_qc.py [hunk 1]
```diff
@@ -425,3 +425,20 @@
...
 pd.read_gbq(query, dialect='standard')
+
+# Checks for basics survey module
+# Participants with data in other survey modules must also have data from the basics survey module.
+# This check identifies survey responses associated with participants that do not have any responses
+# associated with the basics survey module.
+# In ideal circumstances, this query will not return any results.
+
+query = f'''SELECT DISTINCT person_id FROM `{project_id}.{new_rdr}.person` 
+WHERE NOT EXISTS (
```
Comment (jp3477) [inline data_steward/analytics/cdr_ops/rdr_export_qc.py:436]:
```suggestion
WHERE person_id NOT IN (
```
I think this is the correct way to find persons who are not selected from the subquery. `WHERE NOT EXISTS` will always return no rows if at least 1 row is returned by the subquery.

**Review by jp3477 — CHANGES_REQUESTED** [review]
Added one suggestion below.

Also, reading through the ticket description, I can't tell if the check should be for persons with no Basics AND at least 1 other survey response (i.e. excluding persons with no survey responses at all). What do you think about that?

File: data_steward/analytics/cdr_ops/rdr_export_qc.py (hunk 1 above)
Comment (dcampbell-vumc) [reply data_steward/analytics/cdr_ops/rdr_export_qc.py:436]: Nice catch! I'll make that change and repush!

**Comment by dcampbell-vumc** [comment]
> Added one suggestion below.
> 
> Also, reading through the ticket description, I can't tell if the check should be for persons with no Basics AND at least 1 other survey response (i.e. excluding persons with no survey responses at all). What do you think about that?

I think so as well. If a participant doesn't have any basics data but has other survey data that needs to be flagged. If the participant has no basics data and no other survey data at all, that's fine because they may have just signed up for the program and nothing has been filled out other than the preliminary stuff.

**Comment by jp3477** [comment]
> > Added one suggestion below.
> > Also, reading through the ticket description, I can't tell if the check should be for persons with no Basics AND at least 1 other survey response (i.e. excluding persons with no survey responses at all). What do you think about that?
> 
> I think so. If a participant doesn't have any basics data but has other survey data that needs to be flagged. If the participant has no basics data and no other survey data at all, that's fine because they may have just signed up for the program and nothing has been filled out other than the preliminary stuff.

Gotcha, that makes sense. I think the query might need another join to `observation` then, in order to find the additional survey records. Currently it is flagging the latter case where he participants have no records.

**Review by jp3477 — CHANGES_REQUESTED** [review]
Thanks for making the last change. As mentioned in my previous comment, I think another join to `observation` is needed to exclude persons with no other survey responses.

**Comment by dcampbell-vumc** [comment]
@jp3477 I validated that the participants generated from the query have survey data but no basics survey data. I also ensured that if a participant has NO survey data, they are not in the list of participants that meet the above criteria. This is ready for another review!

**Review by jp3477 — APPROVED** [review]
Thanks for the changes. Good to go!

**Pull request merged**